canopy_bioemi_mod.F90

```diff
@@ -0,0 +1,199 @@
+            end if
+        end do
+
+        TLEAF_SUN   = ATEMP_SUN + (BTEMP_SUN*TEMP2)
+        TLEAF_SHADE = ATEMP_SHADE + (BTEMP_SHADE*TEMP2)
+        TLEAF_AVE = (TLEAF_SUN*RJCF) + (TLEAF_SHADE*(1.0-RJCF)) ! average = sum sun and shade weighted by sunlit fraction
+
+! Calculate maximum normalized emission capacity (E_OPT) and Tleaf at E_OPT
+        TLEAF240_AVE   = TLEAF_AVE  !Assume instantaneous TLEAF estimate to get TLEAF240 and TLEAF24 (improve...)
+        TLEAF24_AVE    = TLEAF_AVE
+        TLEAF_OPT = 313.0_rk + (0.6_rk * (TLEAF240_AVE-297.0_rk)) !Guenther et al. (2012)
+
+! Calculate emission species/plant-dependent mapped emission factors
+        call canopy_biop(EMI_IND, LU_OPT, VTYPE, EF, CT1, CEO)
+
+        E_OPT = CEO * EXP(0.05_rk * (TLEAF24_AVE-297.0_rk)) * EXP(0.05_rk * (TLEAF240_AVE-297.0_rk))
+
+! Calculate gamma (activity) values for average Tleaf (Clifton et al., 2022)
+        GammaTLEAF_SUN_NUM = CT2*exp((CT1/(rgasuniv/1000.0))*((1.0/TLEAF_OPT)-(1.0/TLEAF_SUN)))
+        GammaTLEAF_SUN_DEN = (CT2-CT1)*(1.0-exp((CT2/(rgasuniv/1000.0))*((1.0/TLEAF_OPT)-(1.0/TLEAF_SUN))))
+        GammaTLEAF_SUN     = E_OPT*(GammaTLEAF_SUN_NUM/GammaTLEAF_SUN_DEN)
+
+        GammaTLEAF_SHADE_NUM = CT2*exp((CT1/(rgasuniv/1000.0))*((1.0/TLEAF_OPT)-(1.0/TLEAF_SHADE)))
+        GammaTLEAF_SHADE_DEN = (CT2-CT1)*(1.0-exp((CT2/(rgasuniv/1000.0))*((1.0/TLEAF_OPT)-(1.0/TLEAF_SHADE))))
+        GammaTLEAF_SHADE     = E_OPT*(GammaTLEAF_SHADE_NUM/GammaTLEAF_SHADE_DEN)
+
+        GammaTLEAF_AVE = (GammaTLEAF_SUN*RJCF) + (GammaTLEAF_SHADE*(1.0-RJCF)) ! average = sum sun and shade weighted by sunlit fraction
+
+! Calculate gamma (activity) values for average PPFD (Clifton et al., 2022)
+        PPFD240_SUN   = PPFD_SUN/2.0  !Clifton et al...halve the instantaneous PPFD estimate to get PPFD240 and PPFD24 (improve...)
+        PPFD240_SHADE = PPFD_SHADE/2.0
+        PPFD24_SUN    = PPFD_SUN/2.0
+        PPFD24_SHADE  = PPFD_SHADE/2.0
+
+        CP_SUN = 0.0468*(PPFD240_SUN**(0.6))*exp(0.005*(PPFD24_SUN-PPFD0_SUN))
+        CP_SHADE = 0.0468*(PPFD240_SUN**(0.6))*exp(0.005*(PPFD24_SHADE-PPFD0_SHADE))
+        ALPHA_P_SUN = 0.004 - 0.0005*log(PPFD240_SUN)
+        ALPHA_P_SHADE = 0.004 - 0.0005*log(PPFD240_SHADE)
+        GammaPPFD_SUN   = CP_SUN*((ALPHA_P_SUN*PPFD_SUN)/SQRT(1.0 + (ALPHA_P_SUN**2.0) * (PPFD_SUN**2.0)))
+        GammaPPFD_SHADE = CP_SHADE*((ALPHA_P_SHADE*PPFD_SHADE)/SQRT(1.0 + (ALPHA_P_SHADE**2.0) * (PPFD_SHADE**2.0)))
+
+        GammaPPFD_AVE = (GammaPPFD_SUN*RJCF) + (GammaPPFD_SHADE*(1.0-RJCF)) ! average = sum sun and shade weighted by sunlit fraction
+
+! Calculate isoprene emissions profile in the canopy
+        EMI_OUT = 0.0_rk  ! set initial emissions profile to zero
+        do i=1, SIZE(ZK)
+            if (ZK(i) .gt. 0.0 .and. ZK(i) .le. FCH) then  ! above ground level and at/below canopy top
+                FLAI(i) = ((FCLAI(i+1) - FCLAI(i)) * LAI)/FCH    !fractional LAI scaled to canopy depth (~ LAD)
```

Comment:
For layerwise LAD you would divide by the layer thickness (δz). LAD is dLAI/dz.

Reply:
Yes, definitely agreed.   I need to rework the  LAD likely.